packages/mobile/src/screens/chat-screen/ChatUserListItem.tsx

```diff
@@ -0,0 +1,165 @@
+import { useCallback } from 'react'
+
+import { chatActions, chatSelectors } from '@audius/common'
+import type { User } from '@audius/common'
+import { Text, View, TouchableOpacity } from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
+
+import IconUser from 'app/assets/images/iconUser.svg'
+import LoadingSpinner from 'app/components/loading-spinner'
+import { ProfilePicture } from 'app/components/user'
+import { UserBadges } from 'app/components/user-badges'
+import { makeStyles } from 'app/styles'
+import { useThemeColors } from 'app/utils/theme'
+
+const { createChat } = chatActions
+const { getCanMessageUser } = chatSelectors
+
+const messages = {
+  followsYou: 'Follows You',
+  followers: 'Followers',
+  cannotMessage: 'Cannot Be Messaged'
+}
+
+const useStyles = makeStyles(({ spacing, palette, typography }) => ({
+  rootContainer: {
+    backgroundColor: palette.white,
+    flexGrow: 1
+  },
+  profilePicture: {
+    height: spacing(18),
+    width: spacing(18)
+  },
+  border: {
+    borderBottomColor: palette.neutralLight4,
+    borderBottomWidth: 1
+  },
+  userContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    padding: spacing(4)
+  },
+  userNameContainer: {
+    flexGrow: 1,
+    display: 'flex',
+    flexDirection: 'column',
+    marginLeft: spacing(2.5),
+    gap: spacing(1)
+  },
+  userName: {
+    fontSize: typography.fontSize.small,
+    fontWeight: 'bold',
+    color: palette.neutral
+  },
+  followContainer: {
+    marginTop: spacing(1),
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between'
+  },
+  handle: {
+    fontSize: typography.fontSize.small,
+    color: palette.neutral
+  },
+  followersContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center'
+  },
+  followersCount: {
+    fontWeight: '800',
+    marginHorizontal: spacing(1),
+    color: palette.neutralLight4,
+    fontSize: typography.fontSize.small
+  },
+  followers: {
+    color: palette.neutralLight4,
+    fontSize: typography.fontSize.small
+  },
+  iconUser: {
+    height: spacing(4),
+    width: spacing(4)
+  },
+  followsYouTag: {
+    fontSize: typography.fontSize.xxs,
+    fontFamily: typography.fontByWeight.heavy,
+    letterSpacing: 0.64,
+    textTransform: 'uppercase',
+    color: palette.neutralLight4,
+    borderWidth: 1,
+    borderRadius: spacing(1),
+    borderColor: palette.neutralLight4,
+    paddingVertical: spacing(1),
+    paddingHorizontal: spacing(2)
+  },
+  dim: {
+    opacity: 0.5
+  }
+}))
+
+type ChatUserListItemProps = {
+  user: User
+}
+
+export const ChatUserListItem = ({ user }: ChatUserListItemProps) => {
+  const styles = useStyles()
+  const palette = useThemeColors()
+  const dispatch = useDispatch()
+  const canMessageUser = useSelector((state) =>
+    getCanMessageUser(state, user.user_id)
```

Comment:
How often do we refresh the info that drives this selector? Just wondering if it would cause an issue if I open this screen and keep it open while the other user blocks me.

Reply:
Oo good call.. should we poll this?
Personally I think it's ok to leave as is since if a user is blocked, we restrict the creation of new chats as well as disabling the textinput for existing chats, on both sides. So even if a user is able to send messages in the case you mentioned, by the next page load they will be restricted. cc @piazzatron @rickyrombo on this and i'll bring it up during the DMs sync tomorrow.

Reply:
good catch @schottra! def don't think we should poll this. let's handle chat creation failures gracefully instead - maybe navigating to that chat screen with the proper "you cannot message this user" text

Reply:
agreed with @rickyrombo

Reply:
Definitely agree we shouldn't poll. I more imagined that we will discover upon attempting to send a message that you aren't allowed to do it and then add them into the block list at that point.
The real question is do we remove the message from the screen? Or mark it as undelivered? I think the experience could be confusing.